File: gogi/storage/vector/models/document_metadata.go
package models

import "time"

type DocumentMetadata struct {
	DocumentID     string            `json:"document_id"`
	IndexName      string            `json:"index_name"`
	Filename       string            `json:"filename"`
	IngestedAt     time.Time         `json:"ingested_at"`
	ChunkCount     int               `json:"chunk_count"`
	PageCount      int               `json:"page_count,omitempty"`
	WordCount      int               `json:"word_count,omitempty"`
	CustomMetadata map[string]string `json:"custom_metadata,omitempty"`
}

File: workers/worker_utils/documents/document_metadata.py
from pydantic import BaseModel, Field
from typing import Optional, Dict
from datetime import datetime


class DocumentMetadata:
    """Metadata tracked for each ingested document (Listing 5.7)."""

    document_id: str
    index_name: str
    filename: str
    ingested_at: datetime = Field(default_factory=datetime.utcnow)
    chunk_count: int = 0
    page_count: Optional[int] = None
    word_count: Optional[int] = None
    custom_metadata: Optional[Dict[str, str]] = None